src/docfx/lib/git/GitRepoInfoProvider.cs

```diff
@@ -2,16 +2,60 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System.Collections.Concurrent;
+using System.Collections.Generic;
 using System.Diagnostics;
 using System.IO;
+using System.Linq;
 
 namespace Microsoft.Docs.Build
 {
     internal class GitRepoInfoProvider
     {
-        private readonly ConcurrentDictionary<string, GitRepoInfo> _cache
+        private readonly ConcurrentDictionary<string, GitRepoInfo> _folderRepoInfocache
             = new ConcurrentDictionary<string, GitRepoInfo>();
 
+        private readonly ConcurrentDictionary<string, List<GitCommit>> _fileCommitsCache;
+
+        private readonly GitUserProfileCache _githubUserProfileCache = null;
+
+        public bool ProfileInitialized => _githubUserProfileCache != null;
+
+        public GitRepoInfoProvider(Docset docset, List<Document> documents)
+        {
+            Debug.Assert(docset != null);
+            Debug.Assert(documents != null);
+
+            var repoRoot = GitUtility.FindRepo(Path.GetFullPath(docset.DocsetPath));
+            var files = documents.Select(d => PathUtility.NormalizeFile(Path.GetRelativePath(repoRoot, Path.GetFullPath(Path.Combine(docset.DocsetPath, d.FilePath))))).ToList();
+            var commitsList = GitUtility.GetCommits(repoRoot, files);
```

Comment:
Lazy evaluate `GetCommits` because it could take a lot of time

Reply:
can we put it in a BuildCommits Pipeline just like build toc map?

---
In reply to: [197008694](https://github.com/dotnet/docfx/pull/2912#discussion_r197008694) [](ancestors = 197008694)